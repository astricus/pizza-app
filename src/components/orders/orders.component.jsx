import React from 'react';
import { CSSTransition } from 'react-transition-group';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import Order from '../order/order.component';

import { selectCurrentUserOrders } from '../../redux/order/order.selectors';

import {
  OrdersPageContainer,
  OrdersPageHeader,
  OrdersPageText,
} from './orders.styles';
import { selectCurrentUser } from '../../redux/user/user.selectors';

export const Orders = ({ orders, currentUser }) => (
  <CSSTransition
    appear={true}
    in={true}
    timeout={300}
    classNames="orders-animation"
  >
    <OrdersPageContainer>
      <OrdersPageHeader>User: {currentUser.displayName}</OrdersPageHeader>
      <OrdersPageText>
        Total orders: {orders ? orders.length : 0}
      </OrdersPageText>
      {orders
        ? orders.map((order, idx) => (
            <Order key={order.id} order={order} idx={idx + 1} />
          ))
        : null}
    </OrdersPageContainer>
  </CSSTransition>
);

const mapStateToProps = createStructuredSelector({
  currentUser: selectCurrentUser,
  orders: selectCurrentUserOrders,
});

export default connect(mapStateToProps)(Orders);
