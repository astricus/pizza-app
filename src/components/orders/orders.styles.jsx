import styled from 'styled-components';

export const OrdersPageContainer = styled.div`
  width: 55%;
  margin: 100px auto 50px;

  @media screen and (max-width: 800px) {
    margin: 70px auto 30px;
    width: 90%;
  }

  &.orders-animation-appear {
    opacity: 0;
  }
  &.orders-animation-appear-active {
    opacity: 1;
    transition: opacity 300ms ease;
  }
`;

export const OrdersPageHeader = styled.h2`
  font-size: 36px;
`;

export const OrdersPageText = styled.p`
  font-size: 16px;
`;
