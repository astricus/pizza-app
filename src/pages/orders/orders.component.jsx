import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import CheckoutItem from '../../components/checkout-item/checkout-item.component';

import {
  selectCartItems,
  selectCartTotal,
  selectDeliveryCost,
} from '../../redux/cart/cart.selectors';

import { placeOrderStart } from '../../redux/cart/cart.actions';
import { setCurrency } from '../../redux/menu/menu.actions';

import {
  CheckoutPageContainer,
  CheckoutHeaderContainer,
  HeaderBlockContainer,
  TotalContainer,
} from './orders.styles';
import { selectCurrencySymbol } from '../../redux/menu/menu.selectors';

export const OrdersPage = ({
  cartItems,
  total,
  deliveryCost,
  currencySymbol,
}) => {
  return (
    <CheckoutPageContainer>
      <CheckoutHeaderContainer>
        <HeaderBlockContainer>
          <span>Product</span>
        </HeaderBlockContainer>
        <HeaderBlockContainer>
          <span>Description</span>
        </HeaderBlockContainer>
        <HeaderBlockContainer>
          <span>Quantity</span>
        </HeaderBlockContainer>
        <HeaderBlockContainer>
          <span>Price</span>
        </HeaderBlockContainer>
        <HeaderBlockContainer>
          <span>Remove</span>
        </HeaderBlockContainer>
      </CheckoutHeaderContainer>
      {cartItems.map((cartItem) => (
        <CheckoutItem key={cartItem.id} cartItem={cartItem} />
      ))}
      <TotalContainer>
        DELIVERY: {currencySymbol}
        {deliveryCost}
      </TotalContainer>
      <TotalContainer>
        TOTAL: {currencySymbol}
        {total}
      </TotalContainer>
    </CheckoutPageContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  cartItems: selectCartItems,
  total: selectCartTotal,
  deliveryCost: selectDeliveryCost,
  currencySymbol: selectCurrencySymbol,
});

const mapDispatchToProps = (dispatch) => ({
  placeOrderStart: (orderData) => dispatch(placeOrderStart(orderData)),
  setCurrency: (currency) => dispatch(setCurrency(currency)),
});

export default connect(mapStateToProps, mapDispatchToProps)(OrdersPage);
