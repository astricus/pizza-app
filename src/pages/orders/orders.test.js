import React from 'react';
import { shallow } from 'enzyme';
import { OrdersPage } from './orders.component';

let wrapper;
beforeEach(() => {
  const mockProps = {
    cartItems: [],
    total: 100,
  };

  wrapper = shallow(<OrdersPage {...mockProps} />);
});

it('should render OrdersPage component', () => {
  expect(wrapper).toMatchSnapshot();
});
