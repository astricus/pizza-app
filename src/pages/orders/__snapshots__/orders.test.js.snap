// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render OrdersPage component 1`] = `<Connect(Orders) />`;
